src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [toParts, newSeparator] = [
+    toFormat.slice(0, -1),
+    toFormat[toFormat.length - 1],
```

Comment:
The `newSeparator` is incorrectly derived from the `toFormat` array. The last element of `toFormat` is being used as a separator, which may not be correct. Consider using a predefined separator or extracting it from a correctly formatted string.